### PR TITLE
Bugfix row render

### PR DESCRIPTION
**Pull request by biohzrddd**
Fixed bug where all Account Transaction rows would be re-rendered on hover of a single Transaction row.

Watch the logged call count in the below movies. Before it goes up by 20+ (however many transactions are loaded) every time the mouse moves one row, after it goes up by 2 (the one being left and the one entered).

Before:
[Screencast_from_07-27-2023_090817_PM.webm](https://github.com/actualbudget/actual/assets/10577752/94ab0d52-ebee-4d16-ab1c-93061483d344)

After:
[Screencast_from_07-27-2023_103038_PM.webm](https://github.com/actualbudget/actual/assets/10577752/5741dca2-0f29-4e11-a91e-4d88dcb19cc6)

Note: there is still an existing re-render bug where selecting one row re-renders all other rows unnecessarily.

### CODE REVIEW

**Review by j-f1 — COMMENTED** [review]
I think I figured out why it is set up the way it is: on `master`, hovering over a row, then scrolling it out of view and back, will have it come back still highlighted. If you do that on this branch no rows will appear highlighted.

**Comment by biohzrddd** [comment]
> I think I figured out why it is set up the way it is: on `master`, hovering over a row, then scrolling it out of view and back, will have it come back still highlighted. If you do that on this branch no rows will appear highlighted.

Must be browser-dependent? Not seeing on Firefox 115.0.2 on either master or this PR. Where are you saying you see that issue?

**Comment by biohzrddd** [comment]
Also, seeing as how we're not even using the hover for anything but changing styles, this may as well be in-line CSS just like every other component. I'll update to use that, which will remove the unnecessary code.

File: packages/desktop-client/src/components/transactions/TransactionsTable.js [hunk 1]
```diff
@@ -835,7 +835,8 @@ const Transaction = memo(function Transaction(props) {
         isPreview && { color: colors.n5, fontStyle: 'italic' },
         _unmatched && { opacity: 0.5 },
       ]}
-      onMouseEnter={() => onHover && onHover(transaction.id)}
+      onMouseEnter={() => !isNew && setHovered(true)}
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/transactions/TransactionsTable.js:838]: Maybe we can start moving to pointer events?
```suggestion
      onPointerEnter={() => !isNew && setHovered(true)}
```

File: packages/desktop-client/src/components/transactions/TransactionsTable.js [hunk 2]
```diff
@@ -835,7 +835,8 @@ const Transaction = memo(function Transaction(props) {
...
+      onMouseLeave={() => !isNew && setHovered(false)}
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/transactions/TransactionsTable.js:839]: Maybe we can start moving to pointer events?
```suggestion
      onPointerLeave={() => !isNew && setHovered(false)}
```

**Review by joel-jeremy — CHANGES_REQUESTED** [review]

**Review by joel-jeremy — APPROVED** [review]

**Comment by MatissJanis** [comment]
👋 There's a small regression introduced with this PR. The transaction selection checkbox is no longer visible when hovering on the table rows.

Could you take a look @biohzrddd ?

**Pull request merged**